threescale_api/defaults.py

```diff
@@ -314,6 +314,16 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return str(self)
 
+    def __eq__(self, other) -> bool:
+        return (
+            isinstance(other, DefaultResource) and
+            self.__class__ == other.__class__ and
+            (
+                self.entity_name == other.entity_name or
+                self.entity_id == other.entity_id
+            )
```

Comment:
This `__eq__` is still not correct. since you are able to compare `service` and for example `application`. entity id might be the same (or even name).

Therefore you need to also add a check for the subclassing, this might be sufficient

```python
if isinstance(other, self.__class__) or isinstance(self, other.__class__): ...
```

Reply:
I believe this code is enough to check that, right?
`self.__class__ == other.__class__`

Reply:
since we are not using the subclassing here and hopefully anybody else who is using our client this condition should be good enough.